include/cl_khr_icd2.h

```diff
@@ -0,0 +1,80 @@
+#if !defined(CL_PLATFORM_UNLOADABLE)
+#define CL_PLATFORM_UNLOADABLE_KHR 0x0921
+#endif
+
+#if defined(CL_ENABLE_LAYERS) && !defined(CL_LAYER_API_VERSION_200)
+#define CL_LAYER_API_VERSION_200 200
+
+typedef cl_int CL_API_CALL
+clDeinitLayer_t(void);
+
+typedef clDeinitLayer_t *pfn_clDeinitLayer;
+#endif //defined(CL_ENABLE_LAYERS) && !defined(CL_LAYER_API_VERSION_200)
+
+#if !defined(CL_ICD2_TAG_KHR)
+#define CL_ICD2_TAG_KHR ((size_t)0x4F50454E434C3331ULL)
```

Comment:
This doesn't fit size_t on 32-bit platforms, potentially causing compiler warnings.

Reply:
Indeed the expected tag on 32 bit platforms would be `0x434C3331`. I will update the PR with a detection for 32 bit platforms.

Reply:
Found the warning on msvc:
`warning C4310: cast truncates constant value`

Reply:
Replaced with:
```C
#if !defined(CL_ICD2_TAG_KHR)
#if INTPTR_MAX == INT32_MAX
#define CL_ICD2_TAG_KHR ((intptr_t)0x434C3331)
#else
#define CL_ICD2_TAG_KHR ((intptr_t)0x4F50454E434C3331)
#endif
```
Hopefully portable enough.